### PR TITLE
feat: Show hints on proposal editor to get verified or to get turbo

### PR DESCRIPTION
**Pull request by ChaituVR**
Test only after merging https://github.com/snapshot-labs/snapshot-hub/pull/925 and https://github.com/snapshot-labs/snapshot-hub/pull/926

### Summary

Closes: https://github.com/snapshot-labs/workflow/issues/134

- Show a message when someone tries to post a proposal over the limit
- Show a message when someone tries to add more than the choices limit
- Show a message when someone tries to add more than the body chars limit
- Fix in apps/ui/src/helpers/validation.ts - Throws an error if we try to add more choices than the limit while keeping all choices empty

### How to test

1.  Update limits in apps/ui/src/helpers/turbo.ts
for example:
```bash
diff --git a/apps/ui/src/helpers/turbo.ts b/apps/ui/src/helpers/turbo.ts
index 18869ff9..57f3753f 100644
--- a/apps/ui/src/helpers/turbo.ts
+++ b/apps/ui/src/helpers/turbo.ts
@@ -1,15 +1,15 @@
 export const MAX_BODY_LENGTH = {
-  default: 10000,
+  default: 1,
   turbo: 40000
 };
 
 export const MAX_CHOICES = {
-  default: 500,
+  default: 5,
   turbo: 1000
 };
 
 export const MAX_1D_PROPOSALS = {
-  default: 20,
+  default: 1,
   turbo: 40
 };
```
When body chars limit is reached:
image

When proposals limit is reached:
image

When choices limit is reached:
image

### CODE REVIEW

**Comment by wa0x6e** [comment]
Text and the left fire icon do not seem to be aligned

File: apps/ui/src/helpers/turbo.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export const MAX_BODY_LENGTH = {
+  default: 10000,
+  turbo: 40000
+};
+
+export const MAX_CHOICES = {
+  default: 500,
```
Comment (bonustrack) [inline apps/ui/src/helpers/turbo.ts:7]: Can you change default to be 10 for now?

File: apps/ui/src/helpers/turbo.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  turbo: 1000
+};
+
+export const MAX_1D_PROPOSALS = {
+  default: 20,
```
Comment (bonustrack) [inline apps/ui/src/helpers/turbo.ts:12]: Maybe we should add unverified space value too

File: apps/ui/src/helpers/turbo.ts (hunk 1 above)
Comment (ChaituVR) [reply apps/ui/src/helpers/turbo.ts:7]: Why? to test?
try running this in the console

```javascript
for (let i = 0; i < 500; i++) {
  document.querySelector(".static .s-base.mb-5 button.rounded-full.border.button.text-skin-link.bg-skin-bg.w-full.flex.items-center.justify-center.space-x-1").click();
}
```

File: apps/ui/src/helpers/turbo.ts (hunk 2 above)
Comment (ChaituVR) [reply apps/ui/src/helpers/turbo.ts:12]: what error message should we show in this case? something that says "verify your space"?

Comment (ChaituVR) [reply apps/ui/src/helpers/turbo.ts:12]: or do we want to show a turbo message if space hits the unverified limit?

Comment (wa0x6e) [reply apps/ui/src/helpers/turbo.ts:12]: We can show the turbo message, so they could jump from unverified to turbo 🤪

Comment (ChaituVR) [reply apps/ui/src/helpers/turbo.ts:12]: 💵 💵 💵 💵 💵 💵 💵 💵

Comment (bonustrack) [reply apps/ui/src/helpers/turbo.ts:12]: We should show a message: `Get your space verified to increase the limit to ${limit}.`

File: apps/ui/src/components/TurboMessage.vue [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { TURBO_URL } from '@/helpers/turbo';
+
+defineProps<{
+  text: string;
+}>();
+</script>
+
+<template>
+  <div
+    class="flex items-center gap-2 mb-3 border bg-purple-300/20 border-purple-300 text-purple-400 rounded-lg px-3 py-2"
```
Comment (bonustrack) [inline apps/ui/src/components/TurboMessage.vue:11]: Gap-1 is enough

```suggestion
    class="flex items-center gap-1 mb-3 border bg-purple-300/20 border-purple-300 text-purple-400 rounded-lg px-3 py-2"
```

**Comment by wa0x6e** [comment]
Beside the few UI issues raised above, everything's working well for me

Daily proposal limit on unverified spaces should be on another PR I think, since not related to turbo
There's also the issue about choices count above limit not showing any error message, but not related to this PR

**Review by wa0x6e — COMMENTED** [review]
Beside the few UI issues raised above, everything's working well for me

Daily proposal limit on unverified spaces should be on another PR I think, since not related to turbo
There's also the issue about choices count above limit not showing any error message, but not related to this PR

**Comment by ChaituVR** [comment]
If proposal limit is reached and If space is not verified:
image

If proposal limit is reached and If space is not turbo but verified:
image

If choice limit is reached and If space is not turbo:
image

if chars limit is reached and If space is not turbo: (char limit is `1` here, just to get screenshot)
image

**Review by wa0x6e — APPROVED** [review]
tAck

**Comment by wa0x6e** [comment]
We're currently showing this hint for onchain spaces, where proposal body is too long (> 10K).

Do not make sense since turbo is not supported yet for onchain spaces

**Comment by bonustrack** [comment]
Sorry I'm changing the design, I played a bit with it locally and figured out that the error and the turbo hint are saying something too similar that it doesn't worth making it separate, instead I propose something more light like this:
image

Others comments:
A: Spacing between the "Add choice" and error message should be `2`.
B: It should use leading-6 as line-height
C: We should show formated number with `,`

For space proposal limit you can do like this:
image
image

**Comment by ChaituVR** [comment]
Continued in https://github.com/snapshot-labs/sx-monorepo/pull/803

**Pull request closed without merging**